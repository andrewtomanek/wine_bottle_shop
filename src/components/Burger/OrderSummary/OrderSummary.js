import React, { Fragment } from "react";
import classes from "./OrderSummary.module.css";
import Button from "../../UI/Button/Button";

const orderSummary = props => {
  const ingredientSummary = Object.keys(props.ingredients).map(igKey => {
    return (
      <p className={classes.OrderItem} key={igKey}>
        <span style={{ textTransform: "capitalize" }}>{igKey}</span>:{" "}
        {props.ingredients[igKey]}
      </p>
    );
  });

  return (
    <div className={classes.OrderSummary}>
      <h3>Your Order</h3>
      <div className={classes.OrderBox}>{ingredientSummary}</div>
      <div className={classes.OrderControls}>
        <p className={classes.OrderItem}>
          Total Price: {props.price.toFixed(2)}
        </p>
        <Button btnType="Danger" clicked={props.purchaseCancelled}>
          CANCEL
        </Button>
        <Button btnType="Success" clicked={props.purchaseContinued}>
          CONTINUE
        </Button>
      </div>
    </div>
  );
};

export default orderSummary;
