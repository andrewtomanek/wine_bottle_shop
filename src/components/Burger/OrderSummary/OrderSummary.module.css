.OrderSummary {
  width: 100%;
  height: 100%;
  display: grid;
  grid-auto-flow: row;
  align-items: center;
  box-shadow: 0 2px 1px #ccc;
  padding: 0;
}

.OrderBox {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(2, minmax(20vw, 1fr));
  grid-template-rows: repeat(auto-fill, minmax(5vh, 1fr));
  align-items: center;
  box-shadow: 0 2px 1px #ccc;
  padding: 0;
}

.OrderControls {
  width: 100%;
  display: grid;
  grid-template-columns: repeat(4, minmax(10vw, 1fr));
  grid-template-rows: repeat(auto-fill, minmax(5vh, 1fr));
  align-items: center;
  box-shadow: 0 2px 1px #ccc;
  padding: 0.1rem 0.2rem;
  margin: 0rem;
}

.OrderItem {
  outline: none;
  color: var(--red);
  font-family: inherit;
  font-size: 1.1rem;
  font-weight: 700;
  padding: 0.3rem 0.5rem;
  margin: 0rem;
}
