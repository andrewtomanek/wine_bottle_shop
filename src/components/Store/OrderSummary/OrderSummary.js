import React from "react";
import classes from "./OrderSummary.module.css";
import Button from "../../UI/Button/Button";

const orderSummary = props => {
  const inventorySummary = Object.keys(props.inventory).map(igKey => {
    return (
      <div className={classes.OrderItem} key={igKey}>
        <span className={classes.OrderName}>{props.listItems[igKey]}</span>
        <span className={classes.OrderPrice}> {props.inventory[igKey]}</span>
      </div>
    );
  });

  return (
    <div className={classes.OrderSummary}>
      <h3 className={classes.OrderTitle}>Vaše objednávka</h3>
      <div className={classes.OrderBox}>{inventorySummary}</div>
      <div className={classes.OrderControls}>
        <Button btnType="Danger" clicked={props.purchaseCancelled}>
          CANCEL
        </Button>
        <p className={classes.OrderPrice}>Cena: {props.price.toFixed(2)}</p>
        <Button btnType="Success" clicked={props.purchaseContinued}>
          CONTINUE
        </Button>
      </div>
    </div>
  );
};

export default orderSummary;
