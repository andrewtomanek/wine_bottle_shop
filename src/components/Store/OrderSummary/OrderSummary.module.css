.OrderSummary {
  grid-area: 2/1/3/1;
  width: 100%;
  height: 100%;
  display: grid;
  grid-auto-flow: row;
  justify-items: center;
  padding: 0;
}

.OrderBox {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  align-content: space-around;
  justify-content: space-around;
  padding: 0.2rem;
}

.OrderControls {
  width: 90%;
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: 1fr;
  align-items: center;
  align-content: space-around;
  justify-content: space-around;
  padding: 1rem 0rem;
  margin: 0rem;
}

.OrderItem {
  display: grid;
  grid-template-columns: 1fr 5vw;
  align-content: space-around;
  justify-content: space-around;
  justify-items: start;
  padding: 0.3rem 1rem;
}

.OrderTitle {
  outline: none;
  color: var(--red);
  font-family: inherit;
  text-align: center;
  font-size: 1.2rem;
  font-weight: 600;
  padding: 0.1rem 0.5rem;
  margin: 0rem;
}

.OrderName {
  outline: none;
  color: var(--red);
  font-family: inherit;
  text-align: center;
  font-size: 1.1rem;
  font-weight: 700;
  padding: 0.3rem 0.5rem;
  margin: 0rem;
}

.OrderPrice {
  outline: none;
  color: var(--red);
  font-family: inherit;
  font-size: 1.1rem;
  font-weight: 700;
  padding: 0.3rem 0.5rem;
  margin: 0rem;
}
