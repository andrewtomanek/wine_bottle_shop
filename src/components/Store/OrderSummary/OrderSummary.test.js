import React from "react";
import { shallow } from "enzyme";
import OrderSummary from "./OrderSummary";

const purchaseCancelled = jest.fn();
const purchaseContinued = jest.fn();

const props = {
  inventory: {
    shopItem1: 1,
    shopItem2: 2,
    shopItem3: 3,
    shopItem4: 4,
    shopItem5: 5,
    shopItem6: 6,
    shopItem7: 7,
    shopItem8: 8,
  },
  listNames: {
    shopItem1: "Solaris",
    shopItem2: "Dornfelder",
    shopItem3: "Agni",
    shopItem4: "Muscaris",
    shopItem5: "Sauvignon",
    shopItem6: "Tramín",
    shopItem7: "Hibernal",
    shopItem8: "Pálava",
  },
  price: 999,
  purchaseCancelled,
  purchaseContinued,
};

describe("Shallow Render OrderSummary component", () => {
  it("renders Heading", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper.find("h3").text()).toContain("Vaše objednávka");
  });

  it("renders Danger button", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper.find({ btnType: "Danger" }).text()).toContain("<button />");
  });

  it("renders Success button", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper.find({ btnType: "Success" }).text()).toContain("<button />");
  });

  it("renders Sum", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper.find("p").find(".OrderPrice").first().text()).toContain(
      "36"
    );
  });

  it("renders price", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper.find(".OrderPrice").last().text()).toContain("999.00");
  });

  it("renders shop items", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper.find(".OrderBox").children()).toHaveLength(8);
  });

  it("renders 2 buttons", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper.find("button")).toHaveLength(2);
  });

  it("Match Snapshot of OrderSummary component", () => {
    const wrapper = shallow(<OrderSummary {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
