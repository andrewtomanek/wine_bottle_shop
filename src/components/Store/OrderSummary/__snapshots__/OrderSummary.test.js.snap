// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow Render OrderSummary component Match Snapshot of OrderSummary component 1`] = `
<div
  className="OrderSummary"
>
  <h3
    className="OrderTitle"
  >
    Vaše objednávka
  </h3>
  <div
    className="OrderBox"
  >
    <div
      className="OrderItem"
      key="shopItem1"
    >
      <span
        className="OrderName"
      >
        Solaris
      </span>
      <span
        className="OrderPrice"
      >
         
        1
      </span>
    </div>
    <div
      className="OrderItem"
      key="shopItem2"
    >
      <span
        className="OrderName"
      >
        Dornfelder
      </span>
      <span
        className="OrderPrice"
      >
         
        2
      </span>
    </div>
    <div
      className="OrderItem"
      key="shopItem3"
    >
      <span
        className="OrderName"
      >
        Agni
      </span>
      <span
        className="OrderPrice"
      >
         
        3
      </span>
    </div>
    <div
      className="OrderItem"
      key="shopItem4"
    >
      <span
        className="OrderName"
      >
        Muscaris
      </span>
      <span
        className="OrderPrice"
      >
         
        4
      </span>
    </div>
    <div
      className="OrderItem"
      key="shopItem5"
    >
      <span
        className="OrderName"
      >
        Sauvignon
      </span>
      <span
        className="OrderPrice"
      >
         
        5
      </span>
    </div>
    <div
      className="OrderItem"
      key="shopItem6"
    >
      <span
        className="OrderName"
      >
        Tramín
      </span>
      <span
        className="OrderPrice"
      >
         
        6
      </span>
    </div>
    <div
      className="OrderItem"
      key="shopItem7"
    >
      <span
        className="OrderName"
      >
        Hibernal
      </span>
      <span
        className="OrderPrice"
      >
         
        7
      </span>
    </div>
    <div
      className="OrderItem"
      key="shopItem8"
    >
      <span
        className="OrderName"
      >
        Pálava
      </span>
      <span
        className="OrderPrice"
      >
         
        8
      </span>
    </div>
  </div>
  <p
    className="OrderPrice"
  >
    Celkem 
    36
     ks
  </p>
  <div
    className="OrderControls"
  >
    <button
      btnType="Danger"
      clicked={[MockFunction]}
    >
      Zrušit
    </button>
    <p
      className="OrderPrice"
    >
      Cena: 
      999.00
    </p>
    <button
      btnType="Success"
      clicked={[MockFunction]}
    >
      Pokračovat
    </button>
  </div>
</div>
`;
